src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/CommunityApiService.kt

```diff
@@ -0,0 +1,24 @@
+package uk.gov.justice.digital.hmpps.hmppsinterventionsservice.service
+
+import org.springframework.http.HttpStatus
+import org.springframework.stereotype.Service
+import org.springframework.web.reactive.function.client.WebClient
+import org.springframework.web.reactive.function.client.WebClientResponseException
+import reactor.core.publisher.Mono
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.model.delius.User
+
+@Service
+class CommunityApiService(private val communityApiWebClient: WebClient) {
+  fun <T> emptyWhenNotFound(exception: WebClientResponseException): Mono<T> = emptyWhen(exception, HttpStatus.NOT_FOUND)
+  fun <T> emptyWhen(exception: WebClientResponseException, statusCode: HttpStatus): Mono<T> =
+    if (exception.rawStatusCode == statusCode.value()) Mono.empty() else Mono.error(exception)
+
+  fun getDetailsByUsername(username: String): User? {
+    return communityApiWebClient.get()
+      .uri("/secure/users/$username/details")
+      .retrieve()
+      .bodyToMono(User::class.java)
```

Comment:
We'll most likely have CRN lookups from the probation offender search (possibly in the UI?), so we may not use this endpoint. Is it ok to remove the community API service until we have its first use case from a story?